test_report/tests/report.rs

```diff
@@ -0,0 +1,97 @@
+use prost::Message;
+use prost_wkt_types::Timestamp;
+use proto::test_context::test_run::TestCaseRunStatus;
+use tempfile::tempdir;
+use test_report::report::{MutTestReport, Status};
+use test_utils::mock_git_repo::setup_repo_with_commit;
+use test_utils::mock_server::MockServerBuilder;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn publish_test_report() {
+    let temp_dir = tempdir().unwrap();
+    let repo_setup_res = setup_repo_with_commit(&temp_dir);
+    assert!(repo_setup_res.is_ok());
+    let set_current_dir_res = std::env::set_current_dir(&temp_dir);
+    assert!(set_current_dir_res.is_ok());
+    let state = MockServerBuilder::new().spawn_mock_server().await;
+    std::env::set_var("TRUNK_PUBLIC_API_ADDRESS", &state.host);
+    std::env::set_var("CI", "1");
+    std::env::set_var("GITHUB_JOB", "test-job");
+    std::env::set_var("TRUNK_API_TOKEN", "test-token");
+    std::env::set_var("TRUNK_ORG_URL_SLUG", "test-org");
+
+    let thread_join_handle = std::thread::spawn(|| {
+        let test_report = MutTestReport::new("test".into());
+        test_report.add_test(
+            Some("1".into()),
+            "test-name".into(),
+            "test-classname".into(),
+            "test-file".into(),
+            "test-parent-name".into(),
+            None,
+            Status::Success,
+            0,
+            0,
+            0,
+            "test-message".into(),
+        );
+        let result = test_report.publish(".".into());
+        assert_eq!(result, true);
+    });
+    let thread_res = thread_join_handle.join();
+    assert!(thread_res.is_ok());
```

Comment:
Is there a particular reason to do this in a separate thread?

Reply:
Yes, limitations of tokio. We create a runtime to run the upload sync.

```
Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.
```

Reply:
I think you need to spin up Tokio on the current thread in the application logic instead of a full on executor. You can see an example of this here:
https://github.com/trunk-io/analytics-cli/blob/0f0b5d7b608ec03caad873c656f7eb373726ba1d/context-py/src/lib.rs#L136-L142

Reply:
Unfortunately using `new_current_thread` does not remove the requirement of doing this work in a separate thread. For the reasons you mentioned, I will still update it to use that.

Reply:
Oh darn. I fear this will leak implementation :/ I wonder if we should just spawn a thread in the application code then